core/blockchain.go

```diff
@@ -1154,7 +1161,11 @@ func (bc *BlockChain) InsertReceiptChain(blockChain types.Blocks, receiptChain [
 			// Write all the data out into the database
 			rawdb.WriteBody(batch, block.Hash(), block.NumberU64(), block.Body())
 			rawdb.WriteReceipts(batch, block.Hash(), block.NumberU64(), receiptChain[i])
-			rawdb.WriteTxLookupEntries(batch, block)
+
+			// We don't write tx lookup indices here because Geth can offer a CLI flag `txlookuplimt`
+			// with which user can choose to drop historical indices data and only keep latest indices.
+			// After the fast sync, we will reconstruct all missing indices even user requires to keep
+			// all historical indices.
```

Comment:
The main concern for not adding tx indices writing logic in fast sync is: user can specify different limit value.

E.g.
* First time user specifies limit as 9K and launch a fast sync
* Before the fast sync finishes, user stops the Geth and restart it with limit as 0
But it's hard for Geth to know that a few historical tx indices actually has been dropped unless we record the limit value in database.

An optional solution for this is:

We only write `TxIndexTail` in `blockchain.FastSyncCommitHead` function. The tail value is the **start block number** of this fast sync(if txlookuplimit=0). Since we can always ensure that tx  indices from **start block number** until povit point exists in database. Regarding the historical indices, let background gorontine fixes these.

But the drawback of this solution is: If the current fast block head is 10K, then we run fast sync with limit=0, and we fetch 1K and fast sync is finished. So we have to reindex the first 10K indices. It's kind of painful. Also it's can be confused for users that fast sync is done, but some historical transacitons are missing(but maybe we can fix it by offering a API `eth.indexing`)